fiftyone/server/mutation.py

```diff
@@ -271,7 +271,8 @@ async def set_view(
             fose.StateUpdate(state=state),
         )
 
-        return result_view._serialize() if result_view else []
+        # is None comparison only
+        return result_view._serialize() if result_view is not None else []
```

Comment:
so you want to call result_view._serialize() if result_view == []?

Reply:
The result view is always a view. The import detail is preventing a `bool(result_view)` which makes a count request (full collection scan)